sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/ContainerRegistryAnonymousAccessCredential.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Containers.ContainerRegistry
+{
+    /// <summary>
+    /// Used internally to indiate the client is intended for use with anonymous access authentication.
+    /// </summary>
+    internal class ContainerRegistryAnonymousAccessCredential : TokenCredential
```

Comment:
Is this type needed? What if we just used null for the credential instead?

Reply:
Yeah, I actually started out with that approach.  We validate that credentials are non-null in BearerTokenAuthenticationPolicy.  I wondered about removing that there, but not all services have the semantics that a null credential means use anonymous access, so we probably want to keep that validation for other services.

Reply:
I wonder if you could make the type do something by encapsulating the functionality of `GetAnonymousAcrAccessTokenAsync` into its `GetToken` implementation. Then it would flow very naturally like any other credential.

Reply:
I like this way of thinking - let me give it some consideration.  Thanks!

Reply:
Per our discussion, `TokenRequestContext` can't be extended, so there isn't a convenient way to pass the required `service` and `scope` parameters into the method that requests the anonymous access token from the ACR STS.  Given this, the two `GetToken` methods on `ContainerRegistryAnonymousAccessCredential` would still need to throw InvalidOperationException, unless we did something like set properties on the credential before making the call to `GetToken`, which feels wrong to me from an object-oriented design perspective since the complete call isn't encapsulated -- it seems error prone given it could be easy to forget to set those values before making the call.  I'll keep this implementation for now, as it feels to be like the less complex/more consistent approach given the current implementation of the ACR auth policy.